src/blob.rs

```diff
@@ -140,9 +140,10 @@ impl<'a> BlobObject<'a> {
         context: &Context,
         src: impl AsRef<Path>,
     ) -> std::result::Result<BlobObject, BlobError> {
-        match src.as_ref().starts_with(context.get_blobdir()) {
-            true => BlobObject::from_path(context, src),
-            false => BlobObject::create_and_copy(context, src),
```

Comment:
Oh, TIL that clippy doesn't like this.  I used to prefer it because it's one less line.

Reply:
i'd also say, this match-true-false is fine as well and a matter of taste only.